src/edu/washington/escience/myria/operator/DbCreateFunction.java

```diff
@@ -0,0 +1,134 @@
+package edu.washington.escience.myria.operator;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.MyriaConstants;
+import edu.washington.escience.myria.MyriaConstants.FunctionLanguage;
+import edu.washington.escience.myria.accessmethod.AccessMethod;
+import edu.washington.escience.myria.accessmethod.ConnectionInfo;
+import edu.washington.escience.myria.expression.evaluate.GenericEvaluator;
+import edu.washington.escience.myria.functions.PythonFunctionRegistrar;
+import edu.washington.escience.myria.storage.TupleBatch;
+/**
+ *
+ *Class for creating user defined functions.
+ *
+ */
+public class DbCreateFunction extends RootOperator {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /** The connection to the database database. */
+  private AccessMethod accessMethod;
+  /** The information for the database connection. */
+  private ConnectionInfo connectionInfo;
+  /** function name.*/
+  private final String name;
+  /** function body.*/
+  private final String binary;
+  /** function description or text.*/
+  private final String description;
+  /** does function return multiple tuples.*/
+  private final Boolean isMultivalued;
+  /** function language.*/
+  private final MyriaConstants.FunctionLanguage lang;
+  /** function output schema.*/
+  private final String outputType;
+
+  /** logger for this class. */
+  private static final org.slf4j.Logger LOGGER =
+      org.slf4j.LoggerFactory.getLogger(GenericEvaluator.class);
+
+  /**
+   * @param child the source of tuples to be inserted.
+   * @param name function name.
+   * @param connectionInfo the parameters of the database connection.
+   * @param outputType output schema for the function
+   * @param isMultivalued does it return multiple tuples?
+   * @param lang function type
+   * @param binary function body (encoded binary string)
+   * @param description function decription, this is kept in the catalog and not sent to workers.
+   */
+  public DbCreateFunction(
+      final Operator child,
+      final String name,
+      final String description,
+      final String outputType,
+      final Boolean isMultivalued,
+      final FunctionLanguage lang,
+      final String binary,
+      final ConnectionInfo connectionInfo) {
```

Comment:
Does this constructor need connectionInfo at all? You already handle the null case in the init method.